### PR TITLE
Deprecate bit shifting by Long value

### PR DESCRIPTION
**Pull request by lrytz**
Re-submission of #6370, review by @Ichoran

### CODE REVIEW

**Comment by sjrd** [comment]
Perhaps the deprecation message could state that adding `.toInt` to the right-hand-side will preserve the behavior?

**Comment by lrytz** [comment]
Thanks, done.

**Comment by Ichoran** [comment]
I'm not sure that this error message makes it clear enough what the "argument" is.  Strictly speaking of course the RHS must be the argument of the `<<` (or whichever) method.  But it's so customary to think of operators as having type `(A, B) => C` that I think the LHS also may seem like the argument.

LGTM otherwise, and alas, I cannot at the moment think of a phrasing that works substantially better.

File: project/GenerateAnyVals.scala [hunk 1]
```diff
@@ -232,6 +234,7 @@ import scala.language.implicitConversions"""
       "@unboxImpl@" -> "???"
     )
     def interpolations = Map(
+      "@article@"   -> (if (this == I) "an" else "a"),
```
Comment (som-snytt) [inline project/GenerateAnyVals.scala]: How clever. Conflicts with https://github.com/scala/scala/pull/6490/files where I seem to have hit a couple of more spots.

**Comment by som-snytt** [comment]
I thought I knew what you were talking about until I read the deprecation message.

"Shifting by a Long argument is deprecated except when the value is a Long. Calling `toInt` on the argument invokes the deprecated behavior."

Just throwing it out there, late local time.

`non-Long` sounds like a famous pianist. And I wasn't sure about `current behavior`, and maybe `preserves` sounds to me like it's a code gen thing.

**Pull request merged**